### PR TITLE
User agent aspect

### PR DESCRIPTION
**Pull request by odedre**
## Proposed Changes

- added `useIsMobile` and `useUserAgent` hooks
- added user agent aspect
- scope page - open sidebar by default on browser only (defaults to closed on mobile and tablet)
- code and compositions tabs - open sidebar by default on browser only (defaults to closed on mobile and tablet)

### CODE REVIEW

File: scopes/ui-foundation/user-agent/index.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export { UserAgentAspect, UserAgentAspect as default } from './user-agent.aspect';
+export type { UserAgentRenderCtx } from './user-agent.ui.runtime';
+export { UserAgentUI } from './user-agent.ui.runtime';
```
Comment (KutnerUri) [inline scopes/ui-foundation/user-agent/index.ts:3]:
```tsx
export type { UserAgentUI }
````

File: scopes/ui-foundation/user-agent/user-agent.ui.runtime.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { UIAspect, UIRuntime, UiUI } from '@teambit/ui';
+import React, { ReactNode } from 'react';
+import UAParser from 'ua-parser-js';
+import { UserAgentProvider } from '@teambit/ui.hooks.use-user-agent';
+import { UserAgentAspect } from './user-agent.aspect';
+
+export type UserAgentRenderCtx = {
```
Comment (KutnerUri) [inline scopes/ui-foundation/user-agent/user-agent.ui.runtime.tsx:7]: no need for export

**Review by KutnerUri — APPROVED** [review]

**Comment by KutnerUri** [comment]
Please verify this works in SSR!

File: scopes/compositions/compositions/compositions.tsx [hunk 3]
```diff
@@ -32,11 +33,10 @@ export function Compositions() {
 
     selectComposition(next);
   }, [component]);
-
-  const [isSidebarOpen, setSidebarOpenness] = useState(component.compositions.length > 0);
+  const isMobile = useIsMobile();
+  const showSidebar = useMemo(() => !isMobile && component.compositions.length > 0, [component.compositions.length]);
```
Comment (KutnerUri) [inline scopes/compositions/compositions/compositions.tsx:37]: useMemo is unnecessary. Cheaper to just calculate than to start caching response

**Review by KutnerUri — APPROVED** [review]

File: scopes/compositions/compositions/compositions.tsx (hunk 3 above)
Comment (odedre) [reply scopes/compositions/compositions/compositions.tsx:37]: fixed

**Pull request merged**